test/src/main/java/org/apache/accumulo/test/compaction/ExternalCompaction_3_IT.java

```diff
@@ -163,42 +184,34 @@ public void testCoordinatorRestartsDuringCompaction() throws Exception {
       Set<ExternalCompactionId> ecids =
           waitForCompactionStartAndReturnEcids(getCluster().getServerContext(), tid);
 
+      ServerContext ctx = getCluster().getServerContext();
+
+      // Wait for all compactions to start
+      Map<ExternalCompactionId,RunningCompactionInfo> originalRunningInfo = null;
+      do {
+        originalRunningInfo = getRunningCompactionInformation(ctx, ecids);
+      } while (originalRunningInfo == null
+          || originalRunningInfo.values().stream().allMatch(rci -> rci.duration == 0));
+
       // Stop the Manager (Coordinator)
       getCluster().getClusterControl().stop(ServerType.MANAGER);
 
       // Restart the Manager while the compaction is running
       getCluster().getClusterControl().start(ServerType.MANAGER);
 
-      ServerContext ctx = getCluster().getServerContext();
+      Map<ExternalCompactionId,RunningCompactionInfo> postRestartRunningInfo =
+          getRunningCompactionInformation(ctx, ecids);
 
-      // Confirm compaction is still running
       int matches = 0;
-      while (matches == 0) {
-        TExternalCompactionList running = null;
-        while (running == null) {
-          try {
-            Optional<HostAndPort> coordinatorHost =
-                ExternalCompactionUtil.findCompactionCoordinator(ctx);
-            if (coordinatorHost.isEmpty()) {
-              throw new TTransportException(
-                  "Unable to get CompactionCoordinator address from ZooKeeper");
-            }
-            running = getRunningCompactions(ctx, coordinatorHost);
-          } catch (TException t) {
-            running = null;
-            Thread.sleep(2000);
-          }
-        }
-        if (running.getCompactions() != null) {
-          for (ExternalCompactionId ecid : ecids) {
-            TExternalCompaction tec = running.getCompactions().get(ecid.canonical());
-            if (tec != null && tec.getUpdates() != null && !tec.getUpdates().isEmpty()) {
-              matches++;
-              assertEquals(TCompactionState.IN_PROGRESS, getLastState(tec));
-            }
-          }
+      for (Entry<ExternalCompactionId,RunningCompactionInfo> post : postRestartRunningInfo
+          .entrySet()) {
+        if (originalRunningInfo.containsKey(post.getKey())) {
+          assertTrue(
+              (post.getValue().duration - originalRunningInfo.get(post.getKey()).duration) > 0);
```

Comment:
Should incementing matches be moved inside the loop?  Wondering if we expect to always see a compaction from before restart or of that is not certain because of timing issues.

```suggestion
              (post.getValue().duration - originalRunningInfo.get(post.getKey()).duration) > 0);
         matches++;
```

Reply:
if matches should stay where it is then maybe it could be dropped because it should end up being the same as `postRestartRunningInfo.size()` after the loop

Reply:
Removed in 8334df6